lib/spandex_datadog/api_server.ex

```diff
@@ -65,11 +67,18 @@ defmodule SpandexDatadog.ApiServer do
   * `:verbose?` - Only to be used for debugging: All finished traces will be logged. Defaults to `false`
   * `:batch_size` - The number of traces that should be sent in a single batch. Defaults to `10`.
   * `:sync_threshold` - The maximum number of processes that may be sending traces at any one time. This adds backpressure. Defaults to `20`.
+  * `:sample_rate` - The percentage of traces to be sent to DataDog, expressed as `[0.0, 1.0]`. Defaults to `1.0` (100%).
   * `:name` - The name the GenServer should have. Currently only used for testing. Defaults to `SpandexDatadog.ApiServer`
   * `:trap_exits?` - Whether or not to trap exits and attempt to flush traces on shutdown, defaults to `false`. Useful if you do frequent deploys and you don't want to lose traces each deploy.
   """
   @spec start_link(opts :: Keyword.t()) :: GenServer.on_start()
   def start_link(opts \\ []) do
+    case Keyword.fetch(opts, :sample_rate) do
+      :error -> nil
+      {:ok, sample_rate} when sample_rate in [0.0, 1.0] -> nil
```

Comment:
🔧 I'm 99% certain that this guard will only pass when the value is **either** `0.0` or `1.0` (and not `0.5` for example). Did you confirm that this works as intended?

Reply:
You're right, I wanted a _range_ there. I'm now finding out that ranges don't support floats either. I'll replace it with good old boolean logic.

Reply:
@alexocode fixed. Have another look.